arch/risc-v/src/common/riscv_common_memorymap.h

```diff
@@ -65,9 +65,15 @@ extern "C"
 #ifndef __ASSEMBLY__
 EXTERN uintptr_t g_idle_topstack;
 
-/* Address of per-cpu idle stack base */
+/* Address of per-cpu idle stack base.
+ *
+ * The element of this array is auto generated by the following rule
+ * when compiling:
+ *
+ * g_cpux_idlestack[i] = _ebss + SMP_STACK_SIZE * i;
+ */
 
-EXTERN const uint8_t *g_cpux_idlestack[CONFIG_SMP_NCPUS];
+EXTERN const uint8_t * const g_cpux_idlestack[CONFIG_SMP_NCPUS];
```

Comment:
but, why not remove g_cpux_idlestack and directly return g_idle_topstack + cpuid * SMP_STACK_SIZE? it's better than to generate the code dynamically.

Reply:
Excellent. I have check the code, and find only a few function use this array. I will update and get rid of `g_cpux_idlestack` completely.

Reply:
Now the generated code is removed and everything looks clear. Thanks for the suggestion from @xiaoxiang781216.
@anchao Can you review it again and share some suggestion?